### PR TITLE
linear_cross_entropy: support linear_bias on the chunked path

### PR DESCRIPTION
**Pull request by pearu**
Extends the chunked `_linear_cross_entropy_batch_chunked` op to compute forward and backward through a `linear_bias` argument, lifting the previous reference-path-only restriction (which fell back with a warning when `options != None` and `linear_bias` was set). Supported for the 2-D case (`out_features == ()`); K-dim chunked + bias still falls back to the reference path.

Addresses https://github.com/pytorch/pytorch/pull/176338#discussion_r2921908840

Implementation assisted by Claude.

----

Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.13.0) (oldest at bottom):
* #186113
* __->__ #185276

### CODE REVIEW

**Comment by slayton58** [comment]
@claude Please do a deep review focused on subtle correctness bugs and regressions that are easy for human reviewers to miss. Prioritize findings over summary. I dont care about anything that isnt a real bug drop all the noise but that, if you there isnt anything to report just say so with a small comment

File: test/test_nn.py [hunk 1]
```diff
@@ -14972,19 +15109,29 @@ def test_linear_cross_entropy_chunked_gradcheck(self, device, acc_policy):
...
             allow_retain_graph=True,
         )
 
-        def f(i, w):
-            return nn.functional.linear_cross_entropy(
-                i, w, target, reduction="mean", options=options,
-            )
-
-        torch.autograd.gradcheck(f, (inp, weight))
+        if bias:
```
Comment (slayton58) [inline test/test_nn.py]: Can this be simplified to a single path:
```
def f(i, w, b=None):
    return nn.functional.linear_cross_entropy(
                    i, w, target, linear_bias=b, reduction="mean", options=options,
                )
inputs = (inp, weight, linear_bias) if linear_bias else (inp, weight)
torch.autograd.gradcheck(f, inputs)
```

**Review by slayton58 — APPROVED** [review]
Minor nit, nothing significant

File: test/test_nn.py (hunk 1 above)
Comment (pearu) [reply test/test_nn.py]: yes, done.
